docs/software_requirements/system_initialization.sdoc

```diff
@@ -10,3 +10,269 @@ IMPORT_FROM_FILE: software_requirements.sgra
 TBD
 
 [/FREETEXT]
+
+[SECTION]
+TITLE: Interface Requirements
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[REQUIREMENT]
+UID: ZEP-127
+STATUS: Draft
+TYPE: Interface
+COMPONENT: System Initialization
+TITLE: Build-time Registration for System Initialization Notifications
+STATEMENT: >>>
+The Zephyr RTOS shall expose an interface to build-time registration for the following system initialization notifications:
+
+- Pre-CPU Architecture Initialization
+- Pre-Kernel Initialization
+- Post-Kernel Initialized
+- Application Initialization
+<<<
+
+[REQUIREMENT]
+UID: ZEP-143
+STATUS: Draft
+TYPE: Interface
+COMPONENT: System Initialization
+TITLE: Build-time Registration for SMP-Initialized Notification
+STATEMENT: >>>
+Where the Symmetric Multi-Processing feature is enabled,
+
+the Zephyr RTOS shall expose an interface to build-time registration for the following system initialization notifications:
+
+- Symmetric Multi-Processing Initialized.
+<<<
+
+[/SECTION]
+
+[SECTION]
+TITLE: Functional Requirements
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[REQUIREMENT]
+UID: ZEP-128
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Sustain hardware watchdog inherited from preceding executable
+STATEMENT: >>>
+Where the Zephyr RTOS inherits an active hardware watchdog that cannot be reconfigured,
+
+the Zephyr RTOS shall sustain the active hardware watchdog.
+<<<
+
+[REQUIREMENT]
+UID: ZEP-129
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Initial CPU stack usage kept within published limits
+STATEMENT: >>>
+The Zephyr RTOS shall use no more than the published amount of the initial CPU stack.
+<<<
```

Comment:
This needs a related requirement for Zephyr Project to specify, per CPU architecture, the amount initial stack space consumed before transitioning to an internally-allocated stack.